### PR TITLE
Refactor: Use background task framework for file processing 

### PR DESCRIPTION
**Pull request by jonahkagan**
Task: #1023 

Migrate file processing to the background task framework. Now there will be only one way that background work is done. Precursor to adding a progress bar for file processing.

Some bonus effects:
- Tasks run automatically in tests
- The file processing code is now type-checked (it wasn't before)
- Code coverage bumps up
- Reprocessing files gets simpler - we just create a new task

I also moved the code for processing the jurisdictions file into the jurisdictions module so that it would match the way we structure code for all other file uploads.

### CODE REVIEW

File: server/api/batch_tallies.py [hunk 1]
```diff
@@ -1,34 +1,34 @@
...
+def process_batch_tallies_file(jurisdiction_id: str):
+    jurisdiction = Jurisdiction.query.get(jurisdiction_id)
+
+    def process() -> None:
         # We only support one contest for batch audits, so we can just take the
         # first contest from the jurisdiction's universe.
-        assert len(jurisdiction.contests) == 1
-        contest = jurisdiction.contests[0]
+        assert len(list(jurisdiction.contests)) == 1
+        contest = list(jurisdiction.contests)[0]
```
Comment (benadida) [inline server/api/batch_tallies.py:31]: are we doing the query twice here? Should we store `list(jurisdiction.contest)` and then work off that?

File: server/migrations/versions/8b6b89bfbc97_backgroundtask.py [hunk 2]
```diff
@@ -18,31 +18,7 @@
...
-    op.create_foreign_key(
-        op.f("round_draw_sample_task_id_fkey"),
-        "round",
-        "background_task",
-        ["draw_sample_task_id"],
-        ["id"],
-        ondelete="cascade",
-    )
-
+def backfill():  # pragma: no cover
```
Comment (benadida) [inline server/migrations/versions/8b6b89bfbc97_backgroundtask.py:21]: I'm confused as to why this existing migration is changing. Is that expected?

**Review by benadida — APPROVED** [review]
a couple of small questions.

File: server/api/batch_tallies.py (hunk 1 above)
Comment (jonahkagan) [reply server/api/batch_tallies.py:31]: sqlalchemy caches relationship query results, so there's no performance hit, it's just for making the typechecker not complain

File: server/migrations/versions/8b6b89bfbc97_backgroundtask.py (hunk 2 above)
Comment (jonahkagan) [reply server/migrations/versions/8b6b89bfbc97_backgroundtask.py:21]: i was just doing a small refactor to increase code coverage, but looking at it again i may have made a small mistake

Comment (jonahkagan) [reply server/migrations/versions/8b6b89bfbc97_backgroundtask.py:21]: just pushed a fix

**Pull request merged**